org.eclipse.m2e.core/src/org/eclipse/m2e/core/embedder/IMavenExecutionContext.java

```diff
@@ -106,4 +125,31 @@ public interface IMavenExecutionContext {
    */
   ProjectBuildingRequest newProjectBuildingRequest();
 
+  /**
+   * Either join the currents thread {@link IMavenExecutionContext} or creates a fresh one if this thread is currently
+   * not executing. Be aware that because of this might return an already executing context, this context can't be
+   * configured in any way, use {@link IMaven#createExecutionContext()} or
+   * {@link IMavenProjectFacade#createExecutionContext()} to create a specific context that could be configured.
+   * 
+   * @return a {@link IMavenExecutionContext} that could be used for execution
+   * @throws CoreException
+   */
+  static IMavenExecutionContext join() throws CoreException {
+    IMavenExecutionContext context = MavenExecutionContext.getThreadContext();
+    if(context == null) {
+      return MavenPlugin.getMaven().createExecutionContext();
+    }
+    return context;
+  }
+
+  /**
+   * This method allows to check if the current thread can join a running execution
+   * 
+   * @return <code>true</code> if the current thread has an attached {@link IMavenExecutionContext} or
+   *         <code>false</code> otherwise.
+   */
+  static boolean canJoin() {
+    return MavenExecutionContext.getThreadContext() != null;
+  }
+
```

Comment:
Seems not used

Reply:
Yes this is for completeness here, so if code want to do the following:

```
if (canJoin()) {
  //if something is running we want to attach
  join().execute(...)
} else {
 //we don't want the global context but can't join, do something special here
}
```

Reply:
This is now used.